src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -4897,12 +4897,21 @@ QList<QgsVectorLayer *> QgsPostgresProvider::searchLayers( const QList<QgsVector
 QList<QgsRelation> QgsPostgresProvider::discoverRelations( const QgsVectorLayer *self, const QList<QgsVectorLayer *> &layers ) const
 {
   QList<QgsRelation> result;
+
+  // Silently skip if this is a query layer or for some obscure reason there are no table and schema name
+  if ( mIsQuery || mTableName.isEmpty() || mSchemaName.isEmpty() )
+  {
+    return result;
+  }
+
+  // Skip less silently if layer is not valid
   if ( !mValid )
   {
-    QgsLogger::warning( "Error getting the foreign keys of " + mTableName + ": invalid connection" );
+    QgsLogger::warning( QStringLiteral( "Error discovering relations of %1: invalid layer" ).arg( mQuery ) );
```

Comment:
Should we translate log messages like this?

Reply:
```suggestion
    QgsLogger::warning( tr( "Error discovering relations of %1: invalid layer" ).arg( mQuery ) );
```